code/network/multimsgs.cpp

```diff
@@ -9114,3 +9115,133 @@ void process_sexp_packet(ubyte *data, header *hinfo)
 
 	sexp_packet_received(received_packet, num_ubytes);
 }
+
+// TODO, calculate how many we can send.
+constexpr ubyte MAX_TURRET_TRACK_CHANGES = 32;
```

Comment:
If my math is right then you should be able to add about 170 turrets per packet. Best to just slave this to the max packet size rather than some magic number.

Inside the sending function set a const variable to `MAX_PACKET_SIZE - 20` (technically the 20 is a magic number too, but it's a small safety guard). Then add turrets until `packet_size` hits this variable or you've added all the turrets.

Reply:
If we had a lot of packets to write, I would suggest making a global `MAX_SAFE_PACKET_SIZE` define any packet could use.  But, this far along I don't see the point.

I think you're right about comparing it to max packet size, because the pack function for the angles is variable as well.

Reply:
There are one or two other examples in the code where it's done this way (though I don't remember them off-hand). MAX_PACKET_SIZE is already the largest safe packet size though, the extra 20 bytes is so that you don't have to get super tricky with the size check.

So you can safely do `while (packet_size < MY_MAX_SIZE) {...}` and not have to worry about the exact size of the data being added. Each turret looks to add about 7 bytes so it should end the `while` loop with 13 bytes to spare. If that makes sense.

Reply:
I think I'm just going to keep it at 20 because in case anyone has to change this packet in the future, it would probably be good to have a good amount of padding.  I don't think it's going to be that often that a ship hits the limit.